src/Server/DarknetServer.ts

```diff
@@ -0,0 +1,84 @@
+import { DnetServer } from "../DarkNet/models/DarknetServerData";
+import { Icon, labIcon } from "../DarkNet/ui/ServerIcon";
+import { Minigames } from "../DarkNet/controllers/ServerGenerator";
+import { BaseServer, IConstructorParams } from "./BaseServer";
+import { constructorsForReviver, IReviverValue } from "../utils/JSONReviver";
+
+export type IDarknetServer = {
+  /** Hostname. Must be unique */
+  hostname: string;
+  /** IP Address. Must be unique */
+  ip: string;
+  /** Flag indicating whether player has admin/root access to this server */
+  hasAdminRights: boolean;
+  /** Flag indicating whether player is currently connected to this server */
+  isConnectedTo: boolean;
+  /** RAM (GB) used. i.e. unavailable RAM */
+  ramUsed: number;
+  /** RAM (GB) available on this server */
+  maxRam: number;
+  /** Name of company/faction/etc. that this server belongs to, not applicable to all Servers */
+  organizationName: string;
+  /** Flag indicating whether this is a purchased server */
+  purchasedByPlayer: boolean;
+  /** Flag indicating whether this server has a backdoor installed by a player */
+  backdoorInstalled?: boolean;
+  /** If the server has a stasis link applied */
+  hasStasisLink: boolean;
+  /** The amount of ram blocked by the server owner */
+  ramBlock: number;
+  /** The model of the server. Similar models have similar vulnerabilites. */
+  modelId: Minigames;
+  /** The generic password prompt for the server */
+  staticPasswordHint: string;
+  /** Data associated with the password hint */
+  passwordHintData?: string;
+  /** The difficulty rating of the server, associated with its original depth in the net */
+  difficulty: number;
+  /** The depth of the server in the net */
+  x: number;
+  /** The charisma skill required to heartbleed the server */
+  requiredCharismaSkill: number;
+};
+
+export class DarknetServer extends BaseServer implements IDarknetServer, DnetServer {
```

Comment:
Ideally, the interface `DnetServer` wouldn't exist because all its fields would be on the public API.

If that is not feasible, option #2 is to (mostly) abolish it: Chances are, we don't actually need it anywhere except using it for specifying constructor params, so call it that: `IConstructorParams`. We have a tradition of using this same name for every class. You can import the BaseServer version with a different name to extend it.

Note that you *don't* actually need to import `IConstructorParams` from here into `DNetServerData`; it's enough to define it `as const`, and structural typing will (should) handle it.

Reply:
I was able to remove it and just have a handful of params hidden from dnet.getServer() as an addition to the type.